agent/engine/docker_task_engine.go

```diff
@@ -1267,6 +1256,13 @@ func (engine *DockerTaskEngine) startContainer(task *apitask.Task, container *ap
 
 		}
 	}
+	if task.IsExecCommandAgentEnabled() {
+		if err := engine.execCmdAgentMgr.Start(engine.ctx, engine.client, task, container, dockerID); err != nil {
```

Comment:
feels like this should be outside the startContainer method - ie. container moving to RUNNING is blocked by starting exec agent here, when in real it does not affect start container flow.

Reply:
This is another weird thing of the engine; actually this doesn't block the start. I've observed that if the `client.StartContainer` call above (line 1200) succeeds, then container will be reported as RUNNING, because then docker reports the start to the client in ecs agent, which is listening for docker events.

Reply:
So none of the lines below line 1204 will block the start of the container